wiring/src/spark_wiring_cellular_printable.cpp

```diff
@@ -80,18 +77,30 @@ float CellularSignal::getQualityValue() const
     return 0.0f;
 }
 
-// TODO: remove once rssi/qual are removed
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 size_t CellularSignal::printTo(Print& p) const
 {
     size_t n = 0;
-    n += p.print((*this).rssi, DEC);
+    n += p.print(this->getStrengthValue(), 2);
     n += p.print(',');
-    n += p.print((*this).qual, DEC);
+    n += p.print(this->getQualityValue(), 2);
     return n;
 }
-#pragma GCC diagnostic pop
+
+bool CellularSignal::isValid() const
+{
+    return (sig_.rat != NET_ACCESS_TECHNOLOGY_NONE &&
+            sig_.rssi != std::numeric_limits<int32_t>::min() &&
+#if (PLATFORM_ID == PLATFORM_ELECTRON_PRODUCTION)
+            (sig_.rat == NET_ACCESS_TECHNOLOGY_GSM) ? true : (sig_.qual != std::numeric_limits<int32_t>::min()));
```

Comment:
It would be more straightforward to write this as `(sig_.rat == NET_ACCESS_TECHNOLOGY_GSM || sig_.qual != std::numeric_limits<int32_t>::min())`.

Please add a comment that ublox modems do not support/correctly report signal quality in the case of GSM (if that's why we have this check here).

Also note that in the current code, an object for which `isValid()` returns `true` can hold an invalid signal quality value and there's no way for an app to detect that without relying on magic constants.